ostd/src/mm/page/cont_pages.rs

```diff
@@ -100,3 +171,21 @@ impl<M: PageMeta> From<ContPages<M>> for Vec<Page<M>> {
         vector
     }
 }
+
+impl<M: PageMeta> Iterator for ContPages<M> {
+    type Item = Page<M>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.range.start < self.range.end {
+            // SAFETY: each page in the range would be a handle forgotten
+            // when creating the `ContPages` object.
+            let page = unsafe { Page::<M>::from_raw(self.range.start) };
```

Comment:
This is correct since the semantic of cont_pages.next(), though weird, consumes the first page and hands the ownership of the first page to the caller.